test/Microsoft.AspNet.Mvc.Core.Test/ControllerActionInvokerTest.cs

```diff
@@ -1203,6 +1203,647 @@ private struct SampleStruct
+                {
+                    context = await next();
+                });
+
+            var resourceFilter2 = new Mock<IResourceFilter>(MockBehavior.Strict);
+            resourceFilter2
+                .Setup(f => f.OnResourceExecuting(It.IsAny<ResourceExecutingContext>()))
+                .Callback<ResourceExecutingContext>((c) =>
+                {
+                    c.Result = expected.Object;
+                });
+
+            var resourceFilter3 = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            var actionFilter = new Mock<IAsyncActionFilter>(MockBehavior.Strict);
+            var resultFilter = new Mock<IAsyncResultFilter>(MockBehavior.Strict);
+
+            var invoker = CreateInvoker(
+                new IFilter[]
+                {
+                    resourceFilter1.Object, // This filter should see the result retured from resourceFilter2
+                    resourceFilter2.Object,
+                    resourceFilter3.Object, // This shouldn't run - it will throw if it does
+                    actionFilter.Object, // This shouldn't run - it will throw if it does
+                    resultFilter.Object // This shouldn't run - it will throw if it does
+                },
+                // The action won't run
+                actionThrows: true);
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            Assert.Same(expected.Object, context.Result);
+            Assert.True(context.Canceled);
+        }
+
+        [Fact]
+        public async Task InvokeAction_InvokesAsyncResourceFilter_InvalidShortCircuit()
+        {
+            // Arrange
+            var message =
+                "If an IAsyncResourceFilter provides a result value by setting the Result property of " +
+                "ResourceExecutingContext to a non-null value, then it cannot call the next filter by invoking " +
+                "ResourceExecutionDelegate.";
+
+            ResourceExecutedContext context = null;
+            var resourceFilter = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>(async (c, next) =>
+                {
+                    // This is not valid.
+                    c.Result = Mock.Of<IActionResult>();
+                    context = await next();
+                });
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter.Object, });
+
+            // Act
+            var exception = await Assert.ThrowsAsync<InvalidOperationException>(invoker.InvokeAsync);
+
+            // Assert
+            Assert.Equal(message, exception.Message);
+        }
+
+        [Fact]
+        public async Task InvokeAction_AuthorizationFilter_ChallengePreventsResourceFiltersFromRunning()
```

Comment:
Added a resource filter to InvokeAction_ExceptionInAuthorizationFilterCannotBeHandledByOtherFilters